plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -188,8 +190,14 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
         if (fromType == REAL && toType instanceof DecimalType toDecimalType) {
             return Optional.of(createRealToDecimalCoercer(toDecimalType));
         }
-        if (fromType instanceof TimestampType && toType instanceof VarcharType varcharType) {
-            return Optional.of(new TimestampCoercer.LongTimestampToVarcharCoercer(TIMESTAMP_NANOS, varcharType));
+        if (fromType instanceof TimestampType) {
```

Comment:
For safety, can we just check tha the from type is NANOS?
```suggestion
        if (fromType == TIMESTAMP_NANOS) {
```

Reply:
But adding a check like this `(fromType == TIMESTAMP_NANOS)` might not perform coercion for MILLIS or MICROS precision right ?